lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -578,28 +598,52 @@ def _fill_nodes_metadata(
         # Give all the VMs some time to boot and then acquire an IP address.
         timeout = time.time() + environment_context.network_boot_timeout
 
+        if self.host_node.is_remote:
+            remote_node = cast(RemoteNode, self.host_node)
+            conn_info = remote_node.connection_info
+            address = conn_info[constants.ENVIRONMENTS_NODES_REMOTE_ADDRESS]
+
         for node in environment.nodes.list():
             assert isinstance(node, RemoteNode)
 
             # Get the VM's IP address.
-            address = self._get_node_ip_address(
+            local_address = self._get_node_ip_address(
                 environment, log, lv_conn, node, timeout
             )
 
             node_port = 22
             if self.host_node.is_remote:
-                self.host_node.tools[Iptables].start_forwarding(10022, address, 22)
-
-                remote_node = cast(RemoteNode, self.host_node)
-                conn_info = remote_node.connection_info
+                self._port_forwarding_lock.acquire()
```

Comment:
You should acquire the lock using the `with` statement.

```python
with self._port_forwarding_lock:
   ...
```

This will ensure the lock is always releases, including when unexpected exceptions occur.